scripts/scoreCard/generateRoundForm/generatePlayerInputs.js

```diff
@@ -45,14 +56,18 @@ const generatePlayerInputs = () => {
         input.addEventListener("blur", (event) => {
           // the majority of the time when a player bids zero they can not earn any bonus points
           if (event.target.value === "0") {
-            const bonusInput = document.getElementById(
-              `${player}-Bonus_${getCurrentStep()}`
-            );
-
-            // if a value is already set don't override it
-            if (!bonusInput.value) {
-              bonusInput.value = "0";
-            }
+            setBonusToZero(player);
+          }
+        });
+      }
+      if (type === "Won") {
+        input.addEventListener("blur", (event) => {
+          const wantedValue = document.getElementById(
+            `${player}-Wanted_${getCurrentStep()}`
+          ).value;
+          // players who didn't get their bet don't bonus points, so set bonus to 0
+          if (wantedValue != event.target.value && event.target.value > 0) {
```

Comment:
```suggestion
          if (wantedValue !== event.target.value && event.target.value > 0) {
```
This should make the code a little bit more reliable in its behavior.